src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.js

```diff
@@ -377,7 +377,7 @@ export const CloudEnvironmentModal = ({
         ]),
         // Cloud environment button
         div({ style: toolButtonDivStyles }, [
-          isSettingsSupported(toolLabel, cloudProvider) && h(Clickable, {
+          isSettingsSupported(toolLabel, cloudProvider, workspace.accessLevel, workspace?.workspace?.createdDate) && h(Clickable, {
```

Comment:
It looks like you expect `workspace` to be defined (`workspace.accessLevel`), so I don't think it makes sense to handle undefined with `workspace?.workspace?.createdDate`.